common/src/main/scala/akka/contrib/persistence/mongodb/MongoPersistence.scala

```diff
@@ -86,48 +100,150 @@ abstract class MongoPersistenceDriver(as: ActorSystem, config: Config) {
 
   private[mongodb] def cappedCollection(name: String)(implicit ec: ExecutionContext): C
 
-  private[mongodb] def ensureIndex(indexName: String, unique: Boolean, sparse: Boolean, fields: (String,Int)*)(implicit ec: ExecutionContext): C => C
+  private[mongodb] def ensureIndex(indexName: String, unique: Boolean, sparse: Boolean, fields: (String, Int)*)(implicit ec: ExecutionContext): C => C
 
   private[mongodb] def closeConnections(): Unit
 
   private[mongodb] def upgradeJournalIfNeeded(): Unit
 
-  private[mongodb] lazy val indexes: Seq[IndexSettings] = Seq(
-    IndexSettings(journalIndexName, unique = true, sparse = false, JournallingFieldNames.PROCESSOR_ID -> 1, FROM -> 1, TO -> 1),
-    IndexSettings(journalSeqNrIndexName, unique = false, sparse = false, JournallingFieldNames.PROCESSOR_ID -> 1, TO -> -1)
-  )
+  private[mongodb] def upgradeJournalIfNeeded(suffix: String): Unit
+  
+
+  /**
+   * retrieve suffix from persistenceId
+   */
+  private[this] def getSuffixFromPersistenceId(persistenceId: String): String = {
```

Comment:
Instead of `suffixBuilderClass != null && !suffixBuilderClass.trim.isEmpty` is there a way to use an `Option` here, possibly with `filterNot(_.trim.isEmpty)`?